webproject/src/main/webapp/js/chat-script.js

```diff
@@ -0,0 +1,60 @@
+var roomID;
+
+$(document).ready(function () {
+    database = firebase.database(); 
+    name = "user " + Math.floor(Math.random() * Math.floor(5));
+    roomID = window.location.search.substring(1);
+
+    firebase.database().ref('messages/' + roomID).on('child_added', function(snapshot) {    
+        var snap = snapshot.val();
+        var html = "<li class='message' id='message-" + snapshot.key + "'>";
+        html += snap.time + " " + snap.user + ": ";
+        if (snap.type == "text") {
+            html += snap.message;
+        } else {
+            html += "<a href=\"" + snap.message + "\"><img src=\"" + snap.message + "\" /></a>";
+        }
+        html += "</li>";
+ 
+        document.getElementById("messages").innerHTML += html;
+    });
+});
+
+function sendMessage() {
+    var message = document.getElementById("message").value;
+    var date = new Date();
+    var time = hours_with_leading_zeroes(date) + ":" + minutes_with_leading_zeroes(date);
+
+    firebase.database().ref('messages/' + roomID ).push().set({
```

Comment:
this is the part where you should modify to send the request to servlet in the next PR.
the code will be similar to how the image part you done above

Reply:
Understood!